replicate/configmap/configmaps.go

```diff
@@ -0,0 +1,273 @@
+package configmap
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mittwald/kubernetes-replicator/replicate/common"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"sort"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Replicator struct {
+	*common.GenericReplicator
+}
+
+// NewReplicator creates a new config map replicator
+func NewReplicator(client kubernetes.Interface, resyncPeriod time.Duration, allowAll bool) common.Replicator {
+	repl := Replicator{
+		GenericReplicator: common.NewGenericReplicator(common.ReplicatorConfig{
+			Kind:         "ConfigMap",
+			ObjType:      &v1.ConfigMap{},
+			AllowAll:     allowAll,
+			ResyncPeriod: resyncPeriod,
+			Client:       client,
+			ListFunc: func(lo metav1.ListOptions) (runtime.Object, error) {
+				return client.CoreV1().ConfigMaps("").List(lo)
+			},
+			WatchFunc: func(lo metav1.ListOptions) (watch.Interface, error) {
+				return client.CoreV1().ConfigMaps("").Watch(lo)
+			},
+		}),
+	}
+	repl.UpdateFuncs = common.UpdateFuncs{
+		ReplicateDataFrom:        repl.ReplicateDataFrom,
+		ReplicateObjectTo:        repl.ReplicateObjectTo,
+		PatchDeleteDependent:     repl.PatchDeleteDependent,
+		DeleteReplicatedResource: repl.DeleteReplicatedResource,
+	}
+
+	return &repl
+}
+
+// ReplicateDataFrom takes a source object and copies over data to target object
+func (r *Replicator) ReplicateDataFrom(sourceObj interface{}, targetObj interface{}) error {
+	source := sourceObj.(*v1.ConfigMap)
+	target := targetObj.(*v1.ConfigMap)
+
+	// make sure replication is allowed
+	logger := log.
+		WithField("kind", r.Kind).
+		WithField("source", common.MustGetKey(source)).
+		WithField("target", common.MustGetKey(target))
+
+	targetVersion, ok := target.Annotations[common.ReplicatedFromVersionAnnotation]
+	sourceVersion := source.ResourceVersion
+
+	if ok && targetVersion == sourceVersion {
+		logger.Debugf("target %s/%s is already up-to-date", target.Namespace, target.Name)
+		return nil
+	}
```

Comment:
Couldn't we also test this in the generic controller?

Reply:
Most likely. Will check.

Reply:
I've gone through this code and I would leave it as-is (and maybe work on generalizing it later on).

There are slight variations in how each of these controllers act -- some check for `IsReplicationPermitted`, others don't. `rolebindings.go` has this in an `if` branch, others don't.

If it's OK for you, I'd leave that as an exercise for the next patch.

Reply:
>If it's OK for you, I'd leave that as an exercise for the next patch.

Fine by me. Thanks for giving this another look. :+1: